File: src/Test/StateMonadFFICheatcode.fs
module Test.StateMonadFFICheatcode

let runStateMonadFFICheatcode (input: obj) =
    let depth = unbox<int> input
    let mutable total = 0
    for _ in 1 .. 20 do
        let mutable state = 0
        for _ in 1 .. depth do state <- state + 1
        total <- total + state
    total :> obj

File: src/Test/StateMonadFFICheatcode.java
    public static final java.util.function.Function<Object, Object> runStateMonadFFICheatcode = Bench.nativeBenchmark(input -> {
        // The shared FFI wrapper supplies depth 60; keep the PureScript test's 20 repetitions.
        int depth = (Integer) input;
        int result = 0;
        for (int i = 0; i < 20; i++) {
            int state = 0;
            for (int j = 0; j < depth; j++) state++;
            result += state;
        }
        return result;
    });
